core/src/com/unciv/ui/components/ZoomableScrollPane.kt

```diff
@@ -261,23 +264,16 @@ open class ZoomableScrollPane(
 
     override fun draw(batch: Batch?, parentAlpha: Float) {
         if (isAutoScrollEnabled && !Gdx.input.isTouched) {
-
-            val posX = Gdx.input.x
-            val posY = Gdx.input.y
-
-            if (posX <= 2f) {
-                scrollX -= 3f
-            } else if (posX >= stage.viewport.screenWidth - 2f) {
-                scrollX += 3f
-            }
-
-            if (posY <= 6f) {
-                scrollY -= 3f
-            } else if (posY >= stage.viewport.screenHeight - 6f) {
-                scrollY += 3f
+            fun getPanDelta(pos: Int, max: Int): Float {
+                if (pos < 8) return (pos - 8) / 6f  // negative!
+                if (pos > max - 8) return (8 - max + pos) / 6f
+                return 0f
             }
```

Comment:
Please use clearer names and add comments.

Currently, I do not understand the logic here:
What is the 6f about? Why would the *delta* include the current position instead of just moving us around by a set "
max - padding" amount? I confuse